api/v1/intrusiondetection_types.go

```diff
@@ -27,8 +27,40 @@ type IntrusionDetectionSpec struct {
 	// Only DeepPacketInspection is supported for this spec.
 	// +optional
 	ComponentResources []IntrusionDetectionComponentResource `json:"componentResources,omitempty"`
+
+	// AnomalyDetectionSpec provides configuration for running AnomalyDetection Component within
+	// IntrusionDetection
+	AnomalyDetectionSpec *AnomalyDetectionSpec `json:"anomalyDetectionSpec,omitempty"`
+}
+
+type AnomalyDetectionSpec struct {
+	// StorageType Sets the type of storage to use for storing Anomaly Detection Models. By default it will use the ephemeral
+	// emptyDir on the node AD will be deployed to. This field sets precedent for all AD Storage related fields and will only
+	// take into effect on Management and Standalone clusters.
+	// default: Ephemeral
+	StorageType StorageType `json:"storageType,omitempty"`
+
+	// StorageClassName will populate the PersistentVolumeClaim.StorageClassName that is used to provision disks for the
+	// Anomaly Detection API pod for model storage. The StorageClassName should only be modified when no LogStorage is currently
+	// active. We recommend choosing a storage class dedicated to Tigera LogStorage only. Otherwise, model retention
+	// cannot be guaranteed during upgrades. See https://docs.tigera.io/maintenance/upgrading for up-to-date instructions.
+	// Default: tigera-anomaly-detection-storage
+	// +optional
+	StorageClassName string `json:"storageClassName,omitempty"`
 }
 
+// StorageType is sets the type of storage to be used for the specified component.
+type StorageType string
+
+const (
+	// Ephermeal storage type sets the ephermeal emptyDir() to be used by the component. Data created in this storage type will
+	// follow the Pod's lifetime and get created and deleted along with the Pod.
+	EphermealStorageType StorageType = "Ephermeal"
+	// Persistent storage type sets the will attempt to query for the StorageClass specified by StorageClassName created for
```

Comment:
I think there is something wrong with this sentence, I don't understand it.

> componenet's

typo